doc/docs/Python_User_Interface.md

```diff
@@ -4156,6 +4156,85 @@ media](FAQ.md#why-does-my-simulation-diverge-if-0).
 
 </div>
 
+---
+<a id="MaterialGrid"></a>
+
+### MaterialGrid
+
+```python
+class MaterialGrid(object):
+```
+
+<div class="class_docstring" markdown="1">
+
+This class is used to specify materials on a discrete Cartesian grid. A class object is passed as the `material`
+argument of a [`GeometricObject`](#GeometricObject) or the `default_material` argument of the `Simulation`
+constructor (similar to a material function).
+
+</div>
+
+
+
+<a id="MaterialGrid.__init__"></a>
+
+<div class="class_members" markdown="1">
+
+```python
+def __init__(self,
+             grid_size,
+             medium1,
+             medium2,
+             design_parameters=None,
+             grid_type='U_DEFAULT',
+             do_averaging=False,
+             beta=0,
+             eta=0.5):
+```
+
+<div class="method_docstring" markdown="1">
+
+Creates a `MaterialGrid` object.
+
+The input are two materials `medium1` and `medium2` which are linearly interpolated at each grid point using
+a NumPy array `design_parameters` of size `grid_size` (a 3-tuple or `Vector3` of integers) with floating-point values in
+the range [0,1] used to define a linear weight (i.e., 0 is `medium1` and 1 is `medium2`). Currently, only two material
+types are supported: (1) frequency-independent isotropic $\varepsilon$ or $\mu$ and (2) `LorentzianSusceptibility`.
+`medium1` and `medium2` must both be the same type.
+
+[Subpixel smoothing](Subpixel_Smoothing.md) can be enabled by specifying `do_averaging=True`. To improve the accuracy
+of the subpixel smoothing, a [level set](https://en.wikipedia.org/wiki/Level_set) of the grid can be generated by applying
+a projection operator to the grid values after Meep has bilinearly interpolated the user-specified grid onto the
+[Yee grid](Yee_Lattice.md). The projection is based on a hyperbolic tangent function and is specified using the parameters `beta` 
+($\beta$: "smoothness" of the turn on) and `eta` ($\eta$: contraction/dilation). The level set provides a general approach for
+defining a *discontinuous* representation of the otherwise continuously varying (via the bilinear interpolation) grid values.
+The subpixel smoothing is based on an adaptive quadrature scheme with properties `subpixel_maxeval` and `subpixel_tol` which
+can be specified using the [`Simulation`](#Simulation) constructor.
+
+Grids which are symmetric (e.g., mirror, rotation) must be explicitly defined. This requires overlapping a given
+`MaterialGrid` object with a symmetrized copy of itself. In this case, the overlapping grid points are combined using
+the method `grid_type` which is one of `"U_MIN"` (minimum of two grid values), `"U_PROD"` (product), `"U_SUM"` (sum),
+`"U_DEFAULT"` (default material at grid point).
```

Comment:
U_DEFAULT doesn't combine material grids — it just takes the topmost material grid.

In general, these U_* options allow you to combine *any* material grids that overlap in space with no intervening objects.